graphql_compiler/compiler/ir_lowering_sql/__init__.py

```diff
@@ -21,6 +22,78 @@ def visitor_fn(expression):
 
     return new_ir_blocks
 
+
+def _find_non_null_columns(schema_info, query_metadata_table):
+    """Find a column for each non-root location that's non-null if and only if the vertex exists."""
+    non_null_column = {}
+
+    # Find foreign keys used
+    for location, location_info in query_metadata_table.registered_locations:
+        for child_location in query_metadata_table.get_child_locations(location):
+            if isinstance(child_location, FoldScopeLocation):
+                raise NotImplementedError()
+
+            edge_direction, edge_name = get_edge_direction_and_name(child_location.query_path[-1])
+            vertex_field_name = '{}_{}'.format(edge_direction, edge_name)
+            edge = schema_info.join_descriptors[location_info.type.name][vertex_field_name]
+
+            # The value of the column used to join to this table is an indicator of whether
+            # the left join was a hit or a miss.
+            non_null_column[child_location.query_path] = edge.to_column
+
+    return non_null_column
+
+
+class ContextColumn(expressions.Expression):
+    """A column drawn from the global context."""
+
+    def __init__(self, vertex_query_path, column_name):
+        """Construct a new ContextColumn."""
+        super(ContextColumn, self).__init__()
```

Comment:
Pass the args to `__init__`, otherwise the printing of this expression will be broken.